### ISSUE TITLE
Add code coverage action

### ISSUE THREAD

**Issue by mbarlow-rmi** [msmasnadi/OPGEEv4#28]
Once #19 is merged, we'll need to add a workflow to upload results to https://coveralls.io. 

There's an existing GH action to implement this (https://github.com/marketplace/actions/coveralls-github-action), but there are some additional configurations to consider.
- update `.coveralls.yml` with the new repo token
- ensure that the default `github.token` is sufficient for our use case
- update the badges in `README.md` to point to the appropriate endpoint

Also, this step should probably run only on PRs into `master` or immediately after a PR is merged into  `master`.
